docs/data/material/experimental-api/css-variables/css-variables.md

```diff
@@ -0,0 +1,219 @@
+  }
+}
+```
+
+### Customizing the theme
+
+If you want, for example, to override Material UI's default color schemes, you can use the `experimental_extendTheme` utility.
+
+```jsx
+const theme = experimental_extendTheme({
+  colorSchemes: {
+    light: {
+      palette: {
+        primary: teal,
+        secondary: deepOrange,
+      },
+    },
+    dark: {
+      palette: {
+        primary: cyan,
+        secondary: orange,
+      },
+    },
+  },
+});
+```
+
+{{"demo": "CssVarsCustomTheme.js", "iframe": true }}
+
+If you are using [`ThemeProvider`](/material-ui/customization/theming/#theme-provider), you can replace it with the new experimental provider.
+
+```diff
+- import { ThemeProvider, createTheme } from '@mui/material/styles';
++ import { Experimental_CssVarsProvider as CssVarsProvider } from '@mui/material/styles';
+
+function App() {
+  return (
+-    <ThemeProvider theme={createTheme()}>
+-      ...
+-    </ThemeProvider>
++    <CssVarsProvider>
++      ...
++    </CssVarsProvider>
+  )
+}
+```
+
+### Toggle between light and dark mode
+
+`Experimental_CssVarsProvider` provides light and dark mode by default.
+It stores the selected user mode and syncs it with the browser's local storage internally.
+You can read and update the mode via the `useColorScheme` API.
+
+```jsx
+import {
+  Experimental_CssVarsProvider as CssVarsProvider,
+  useColorScheme,
+} from '@mui/material/styles';
+
+const ModeSwitcher = () => {
+  const { mode, setMode } = useColorScheme();
+  const [mounted, setMounted] = React.useState(false);
+
+  React.useEffect(() => {
+    setMounted(true);
+  }, []);
+
+  if (!mounted) {
+    // for server-side rendering
+    // Read more on https://github.com/pacocoursey/next-themes#avoid-hydration-mismatch
+    return null;
+  }
+
+  return (
+    <Button
+      variant="outlined"
+      onClick={() => {
+        if (mode === 'light') {
+          setMode('dark');
+        } else {
+          setMode('light');
+        }
+      }}
+    >
+      {mode === 'light' ? 'Dark' : 'Light'}
+    </Button>
+  );
+};
+
+function App() {
+  return (
+    <CssVarsProvider>
+      <ModeSwitcher />
+    </CssVarsProvider>
+  );
+}
+```
+
+### Server-side rendering
+
+To prevent the dark-mode SSR flickering during the hydration phase, place `getInitColorSchemeScript()` before the `<Main />` tag.
+
+#### Next.js
+
+To use the API with a Next.js project, add the following code to the `pages/_document.js` file.
+
+```jsx
+import Document, { Html, Head, Main, NextScript } from 'next/document';
+import { getInitColorSchemeScript } from '@mui/material/styles';
+
+export default class MyDocument extends Document {
+  render() {
+    return (
+      <Html>
+        <Head>...</Head>
+        <body>
+          {getInitColorSchemeScript()}
+          <Main />
+          <NextScript />
+        </body>
+      </Html>
+    );
+  }
+}
+```
+
+#### Gatsby
+
+To use the API with a Gatsby project, add the following code to the `gatsby-ssr.js file`
```

Comment:
Same idea as the Next.js suggestion, since this file is something that's not included in a blank Gatsby app.

```suggestion
To use the API with a Gatsby project, add the following code to the custom [`gatsby-ssr.js](https://www.gatsbyjs.com/docs/reference/config-files/gatsby-ssr/) file`:
```